### PR TITLE
Add workaround for KtLint U+FEFF issue

### PR DESCRIPTION
**Pull request by MattWindsor91**
Works around the issue described in [this KtLint issue](https://github.com/pinterest/ktlint/issues/3220).

The workaround is simple if inelegant: we just always make sure there's a BOM before things go into KtLint, thereby forcing KtLint onto the happy path where it sees the BOM, removes it, and adds it back in.  We then remove the BOM again, if we were the ones who added it.  (So, writ large, we add a BOM, KtLint removes it, KtLint puts it back, we remove it again...)

It's worth noting that we use an old version of KtLint (1.4.0), but the bug seems to be still in place in 1.8.0.

### CODE REVIEW

File: azuki-script-generation/src/test/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationServiceBomTest.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * Tests some BOM corner cases in script generation and formatting.
+ */
+@RunWith(Parameterized::class)
+class ScriptGenerationServiceBomTest(
+    private var name: String, private var formatter: Formatter, private var prefix: String
+) {
+
+    @Test
+    fun test() {
+
```
Comment (donbex) [inline azuki-script-generation/src/test/kotlin/com/anaplan/engineering/azuki/script/generation/ScriptGenerationServiceBomTest.kt:17]: ⛏️ 
```suggestion
```

**Review by donbex — APPROVED** [review]
